index.js

```diff
@@ -1,50 +1,25 @@
 module.exports = merge
 
 function merge() {
-  var objs = [].slice.call(arguments)
-    , l = objs.length
-    , i = 0
-
-  if(!l) return {}
-  if(l === 1) return objs[0]
-
-  var result = {}
-    , keys = []
-
-  var keyResult
-    , convert
-    , attr
-    , el
-
-  for(; i < l; ++i) {
-    keys = keys.concat(Object.keys(objs[i]))
-  }
-
-  for(i = 0, l = keys.length; i < l; ++i) {
-    if(keys.lastIndexOf(keys[i]) !== i) {
-      keys.splice(i, -1)
+  var objs = [].slice.call(arguments),
+      len = objs.length
+
+  if(!len) return {}
+  if(len === 1) return objs.shift()
+
+  function mergeObjects(curr, next) {
+    var keys = Object.keys(curr),
+        key
+
+    for(var i = 0, len = keys.length; i < len; ++i) {
+      key = keys[i]
+      if(next.hasOwnProperty(key)) 
+        next[key] = [].concat(curr[key]).concat(next[key])
+      else 
+        next[key] = curr[key]
     }
+    return next
   }
 
-  for(i = 0, l = keys.length; i < l; ++i) {
-    keyResult = []
-    attr = keys[i]
-    convert = false
-
-    for(var j = 0, k = objs.length; j < k; ++j) {
-      el = objs[j]
-      if(!el.hasOwnProperty(attr)) continue
-      if(!Array.isArray(el[attr])) {
-        keyResult.push(el[attr])
-      } else {
-        keyResult = keyResult.concat(el[attr])
-        convert = true
-      }
-    }
-
-    if(keyResult.length === 1 && !convert) keyResult = keyResult[0]
-    result[attr] = keyResult
-  }
-
-  return result
+  return objs.reduce(mergeObjects)
```

Comment:
would rather this statement went above the function declaration for `mergeObjects`